src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,974 @@
+
+            Iterable<? extends JavaFileObject> compilationUnits = fileManager
+                    .getJavaFileObjectsFromFiles(javaFilesChanged);
+            JavaCompiler.CompilationTask task = compiler.getTask(null, fileManager, null, optionList, null,
+                    compilationUnits);
+            boolean didCompile = task.call();
+            if (didCompile) {
+                if (tests) {
+                    info("Tests compilation was successful.");
+                } else {
+                    info("Source compilation was successful.");
+                }
+
+                // run tests after successful compile
+                if (tests) {
+                    // if only tests were compiled, don't need to wait for
+                    // app to update
+                    runTestThread(false, executor, -1, false, false);
+                } else {
+                    runTestThread(true, executor, messageOccurrences, false, false);
+                }
+            } else {
+                if (tests) {
+                    info("Tests compilation had errors.");
+                } else {
+                    info("Source compilation had errors.");
+                }
+            }
+        } catch (Exception e) {
+            debug("Error compiling java files", e);
+        }
+    }
+
+    /**
+     * Gets the class path for the specified artifactPaths and outputDirs.
+     * 
+     * @param artifactPaths list of artifacts for the current project
+     * @param outputDirs list of output directories for the current project
+     * @return set of classpath files
+     * @throws IOException unable to resolve canonical path
+     */
+    protected Set<File> getClassPath(List<String> artifactPaths, List<File> outputDirs) throws IOException {
+        List<URL> urls = new ArrayList<>();
+        ClassLoader c = Thread.currentThread().getContextClassLoader();
+        while (c != null) {
+            if (c instanceof URLClassLoader) {
+                urls.addAll(Arrays.asList(((URLClassLoader) c).getURLs()));
+            }
+            c = c.getParent();
+        }
+
+        Set<String> parsedFiles = new HashSet<>();
+        Deque<String> toParse = new ArrayDeque<>();
+        for (URL url : urls) {
+            toParse.add(new File(url.getPath()).getCanonicalPath());
+        }
+
+        for (String artifactPath : artifactPaths) {
+            toParse.add(new File(artifactPath).getCanonicalPath());
+        }
+
+        Set<File> classPathElements = new HashSet<>();
+        classPathElements.addAll(outputDirs);
+        while (!toParse.isEmpty()) {
+            String s = toParse.poll();
+            if (!parsedFiles.contains(s)) {
+                parsedFiles.add(s);
+                File file = new File(s);
+                if (file.exists() && file.getName().endsWith(".jar")) {
+                    classPathElements.add(file);
+                    if (!file.isDirectory() && file.getName().endsWith(".jar")) {
```

Comment:
You already checked that it ends with ".jar" at line 897. This check here seems unnecessary.

Reply:
fixed in https://github.com/WASdev/ci.common/pull/52